### ISSUE TITLE
Impossible to reuse XbaseProposalProvider.getProposalFactory

### ISSUE THREAD

**Issue by OLibutzki** [eclipse/xtext-eclipse#365]
I have an Xbase based language and would like to customize the ProposalFactory. I could override org.eclipse.xtext.xbase.ui.contentassist.XbaseProposalProvider.getProposalFactory(String, ContentAssistContext), but it's impossible to extend the ProposalCreator returned by XbaseProposalProvider.getProposalFactory as it's an anonymous class.

It would be better to have a XbaseProposalCreator which extends DefaultProposalCreator. This solution provides the possibility to extend XbaseProposalCreator and fallback to its implementation.

**Comment by OLibutzki** [comment]
Ok, I could use delegation instead:
```
override protected getProposalFactory(String ruleName, ContentAssistContext contentAssistContext) {
	val superProposalFactory = super.getProposalFactory(ruleName, contentAssistContext)
	return new DefaultProposalCreator(contentAssistContext, ruleName, getQualifiedNameConverter()) {
		
		override apply(IEObjectDescription candidate) {
			// might create a proposal
			// [...]
			// fallback
			return superProposalFactory.apply(candidate)
		}
	}
}
```
Anyway, implementing an anonymous class with much more than 100 lines of code might not be the best solution.

**Comment by cdietrich** [comment]
DefaultProposalCreator is a protected inner class so it wont be visible => XbaseProposalCreator would need to be a inner class as well

**Comment by OLibutzki** [comment]
@cdietrich exactly

**Comment by cdietrich** [comment]
fixed in Xtext 2.13
